src/_guides/libraries/testing.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: default
+title: "Dart Testing"
+description: "How to test Dart, Flutter, and 'Dart for the Web' applications?"
+---
+
+Software testing, an important part of app development, helps verify that
+your app is working correctly before unleashing it on your users.
+This Dart testing guide outlines several types of testing, and points
+you to where you can learn more about testing on the Flutter, Dart VM,
+and 'Dart for the Web' platforms.
+
+## Testing types
+
+The Dart testing docs focus on three types of testing, out of the
+[many testing types](http://www.aptest.com/testtypes.html) that
+you may be familiar with: unit, widget (or component, depending on the
+platform), and integration (or end-to-end). Testing terminology varies,
+but these are the terms and concepts that you are likely to
+encounter when using Dart technologies:
+
+* _Unit_ tests focus on verifying the smallest piece of testable
+  software, such as a function, method, or class. Your testing suite
+  should have more unit tests that widget or integration tests.
+
+* _Widget_ (or _component_) tests verify that a widget (which
```

Comment:
How about:
> * _Component_ / _widget_ tests verify that a component or widget (implemented from one or more classes) behaves as expected.

I.e., somehow it would be nice if we could put the terms "component" and "widget" on equal footing.